src/ristretto/pedersen/mod.rs

```diff
@@ -0,0 +1,165 @@
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{borrow::Borrow, iter::Sum};
+
+use curve25519_dalek::{
+    constants::{RISTRETTO_BASEPOINT_COMPRESSED, RISTRETTO_BASEPOINT_POINT},
+    ristretto::{CompressedRistretto, RistrettoPoint},
+};
+
+use crate::{
+    commitment::HomomorphicCommitment,
+    ristretto::{
+        constants::{RISTRETTO_NUMS_POINTS, RISTRETTO_NUMS_POINTS_COMPRESSED},
+        RistrettoPublicKey,
+    },
+};
+
+pub mod commitment_factory;
+pub mod extended_commitment_factory;
+
+pub const RISTRETTO_PEDERSEN_G: RistrettoPoint = RISTRETTO_BASEPOINT_POINT;
+pub const RISTRETTO_PEDERSEN_G_COMPRESSED: CompressedRistretto = RISTRETTO_BASEPOINT_COMPRESSED;
+lazy_static! {
+    pub static ref RISTRETTO_PEDERSEN_H: RistrettoPoint = RISTRETTO_NUMS_POINTS[0];
+    pub static ref RISTRETTO_PEDERSEN_H_COMPRESSED: CompressedRistretto = RISTRETTO_NUMS_POINTS_COMPRESSED[0];
+}
+
+pub type PedersenCommitment = HomomorphicCommitment<RistrettoPublicKey>;
+
+impl<T> Sum<T> for PedersenCommitment
+where T: Borrow<PedersenCommitment>
+{
+    fn sum<I>(iter: I) -> Self
+    where I: Iterator<Item = T> {
+        let mut total = RistrettoPoint::default();
+        for c in iter {
+            let commitment = c.borrow();
+            total += RistrettoPoint::from(&commitment.0);
+        }
+        let sum = RistrettoPublicKey::new_from_pk(total);
+        HomomorphicCommitment(sum)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use bulletproofs_plus::generators::pedersen_gens::ExtensionDegree;
+    use tari_utilities::ByteArray;
+
+    use crate::{
+        commitment::{ExtendedHomomorphicCommitmentFactory, HomomorphicCommitmentFactory},
+        keys::{PublicKey, SecretKey},
+        ristretto::{
+            pedersen::{
+                commitment_factory::PedersenCommitmentFactory,
+                extended_commitment_factory::ExtendedPedersenCommitmentFactory,
+                PedersenCommitment,
+                RISTRETTO_PEDERSEN_G,
+                RISTRETTO_PEDERSEN_H,
+            },
+            RistrettoPublicKey,
+            RistrettoSecretKey,
+        },
+    };
+
+    static EXTENSION_DEGREE: [ExtensionDegree; 6] = [
```

Comment:
DRY - Move the test that uses this to `extended_commitment_factory.rs` and then you don't have to duplicate this code.

Reply:
I have added this test at the module level as it logically belongs there, but move the code as suggested in favour of DRY.